yktrace/src/swt.rs

```diff
@@ -121,36 +122,68 @@ unsafe fn yk_swt_start_tracing_impl() {
     }
 
     TRACE_BUF_CAP = TL_TRACE_INIT_CAP;
-    TRACING = true;
+    __YK_SWT_ACTIVE = true;
 }
 
 /// Record a location into the trace buffer if tracing is enabled on the current thread.
 #[do_not_trace]
 #[no_mangle]
+#[cfg(not(all(target_arch = "x86_64", target_os = "linux")))]
 unsafe extern "C" fn __yk_swt_rec_loc(symbol_name: *const i8, bb_idx: u32) {
     if !TRACING {
         return;
     }
 
+    __yk_swt_rec_loc_impl(symbol_name, bb_idx);
+}
+
+#[cfg(all(target_arch = "x86_64", target_os = "linux"))]
+global_asm!(r#"
+.globl __yk_swt_rec_loc
+.type __yk_swt_rec_loc,@function
+.section .text.__yk_swt_rec_loc,"ax",@progbits
+__yk_swt_rec_loc:
+    push %rdi
+    data16 lea __YK_SWT_ACTIVE@tlsgd(%rip),%rdi
```

Comment:
What's the purpose of `data16` here?

Reply:
Is the `@tlsgd` suffix guaranteed to be fixed?

Reply:
Ah, the `data16` is like an `.asciz` in gnu asm?

Reply:
i.e. inlining data.

Reply:
The block of assembly must result in an exact combination of bytes for the linker to be able to rewrite the expensive `__tls_get_addr` call to a sequence using `%fs:0` and a fixed offset. If you fail to use this exact block of assembly you get a linker error.

Reply:
Sorry for being a dummy, but can you add a comment explaining about why `tls_get_addr` is expensive?

Reply:
`__tls_get_addr` walks a linked list containing the tls data for every object linked into the current process.

Reply:
Gotcha. That's a detail that should appear in a comment :)